src/Simulator/FluidSimulator.cpp

```diff
@@ -128,100 +128,96 @@ bool cFluidSimulator::IsHigherMeta(NIBBLETYPE a_Meta1, NIBBLETYPE a_Meta2)
 
 
 
-// TODO Not working very well yet :s
-Direction cFluidSimulator::GetFlowingDirection(int a_X, int a_Y, int a_Z, bool a_Over)
+Vector3f cFluidSimulator::GetFlowingDirection(int a_X, int a_Y, int a_Z)
 {
+	Vector3f vDirection;
```

Comment:
This variable isn't actually used until line 208. The other cases would be clearer as `return {};` for the zero vector.
Also remember to drop the prefix and just call this `Direction`.